### DISCUSSION TITLE
Why the classic columns like `team_id` on table `players` are now called `player_team`?

### DISCUSSION THREAD

**Discussion by frederikhors** [ent/ent#2615] [Q&A] [answered]
I'm new to Ent and I'm designing the first relations in these minutes.

What I don't understand (from docs too) is why the classic columns like `team_id` on table `players` are now called like `player_team`.

Is this because I'm designing the edges wrong or is it your choice? **And why this choice?**

_Excuse my curiosity, but I want to learn from who knows more than me and who is smarter than me._

**Comment by a8m** [comment, marked as answer]
Hey @frederikhors 👋 

The reason for it explain here: https://github.com/ent/ent/issues/548#issuecomment-643636090. However, you can control this name (and expose this field) using the [edge field](https://entgo.io/docs/schema-edges#edge-field) **or** storage-key options.

**Reply by frederikhors** [reply]
Ok.

But I found unuseful and DRY to repeat the column name multiple times. I really would like to have `team_id` automagically generated.

**Reply by a8m** [reply]
Based on my referenced comments above, we can’t automatically generate these fields. I also don’t want to expose them in some cases, and I prefer the opt-in approach (rather than opt-out) because we stick with the "explicit way" in Ent.

If you want to generate these edge fields, inject them into the graph using codegen hooks.
